knowledge-base/gantt-tooltip-access-model-fields.md

```diff
@@ -0,0 +1,274 @@
+                     Width="60px"
+                     DisplayFormat="{0:P}">
+        </GanttColumn>
+        <GanttColumn Field="Start"
+                     Width="100px"
+                     DisplayFormat="{0:d}">
+        </GanttColumn>
+        <GanttColumn Field="End"
+                     Width="100px"
+                     DisplayFormat="{0:d}">
+        </GanttColumn>
+    </GanttColumns>
+    <GanttViews>
+        <GanttWeekView></GanttWeekView>
+    </GanttViews>
+</TelerikGantt>
+
+@code {
+    private List<FlatModel> Data { get; set; }
+
+    private FlatModel GetModel(string title)
+    {
+        return Data.FirstOrDefault(x => x.Title == title);
+    }
+
+    public class FlatModel
+    {
+        public int Id { get; set; }
+        public int? ParentId { get; set; }
+        public string Title { get; set; }
+        public double PercentComplete { get; set; }
+        public DateTime Start { get; set; }
+        public DateTime End { get; set; }
+        public string CustomField { get; set; }
+        public int SomeIntField {get; set;}
+    }
+
+    public int LastId { get; set; } = 1;
+
+    protected override void OnInitialized()
+    {
+        Data = new List<FlatModel>();
+        var random = new Random();
+
+        for (int i = 1; i < 6; i++)
+        {
+            var newItem = new FlatModel()
+                {
+                    Id = LastId,
+                    Title = "Task  " + i.ToString(),
+                    Start = new DateTime(2021, 7, 5 + i),
+                    End = new DateTime(2021, 7, 11 + i),
+                    PercentComplete = Math.Round(random.NextDouble(), 2),
+                    CustomField = "Details for task " + i, 
+                    SomeIntField = i
+                };
+
+            Data.Add(newItem);
+            var parentId = LastId;
+            LastId++;
+
+            for (int j = 0; j < 5; j++)
+            {
+                Data.Add(new FlatModel()
+                    {
+                        Id = LastId,
+                        ParentId = parentId,
+                        Title = "    Task " + i + " : " + j.ToString(),
+                        Start = new DateTime(2021, 7, 5 + j),
+                        End = new DateTime(2021, 7, 6 + i + j),
+                        PercentComplete = Math.Round(random.NextDouble(), 2),
+                        CustomField = "Details for task " + i,
+                        SomeIntField = j
+                    });
+
+                LastId++;
+            }
+        }
+
+        base.OnInitialized();
+    }
+}
+````
+
+### Use a Custom Tooltip
+
+This solution relies on disabling the built-in Tooltip and adding a custom one, so you have full control over its rendering and content.
+
+1. Disable the built-in Tooltip - set the `ShowTooltip` parameter to `false`.
+1. Use [`TaskTemplate`]({%slug gantt-task-template%}) to add a unique `id` to the task.
+1. Add a [TelerikTooltip]({%slug tooltip-overview%}) in the `TaskTemplate`.
```

Comment:
The better option is to use a single Tooltip instance for the whole Gantt.
https://docs.telerik.com/blazor-ui/knowledge-base/tooltip-in-grid

Reply:
In an offline discussion, agreed to keep the example as is because the clients can directly get the model from the TaskTemplate context. Added a note.